include/cache.php

```diff
@@ -84,9 +84,14 @@ public static function get($key) {
 		$memcache = self::memcache();
 		if (is_object($memcache)) {
 			// We fetch with the hostname as key to avoid problems with other applications
-			$value = $memcache->get(get_app()->get_hostname().":".$key);
-			if (!is_bool($value)) {
-				return unserialize($value);
+			$cached = $memcache->get(get_app()->get_hostname().":".$key);
+			$value = @unserialize($cached);
+
+			// Only return a value if the serialized value is valid.
+			// We also check if the db entry is a serialized
+			// boolean 'false' value (which we want to return).
+			if ($cached === 'b:0;' || $value !== false) {
+				return $value;
```

Comment:
I have read that `unserialize()` does not throw Exception. From my understand we would need to set up a personal Exception handler. From my understanding about the topic this personal Exception handler can be used also at other places. For me some questions arose (which I wanted to ask at the hackathon). At the moment I don't have the time to step into a new topic and for doing some tests.  I fear this is a sentence that you will hear the next eight months more often. :-/  (I'm out of money and need to finish university as fast as possible)